### DISCUSSION TITLE
TypeError: object of type 'NoneType' has no len()

### DISCUSSION THREAD

**Discussion by annieM459** [earthlab-education/Earth-Analytics-2023-01-Intro#618] [Q&A08 Multispectral Data + Loops + Functions + Conditionals]
I am using shapefiles for the channel islands for my mutlispectral on your own work.  However, I recieve this error when I try to download the data and process rhe raster. Does this mean the data simply doesn't exist for the islands? 
<img width="1051" alt="Screenshot 2023-11-24 at 5 44 17 PM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/140002702/6a3c7682-2744-4144-b939-2cbd2a56b434">

**Comment by eculler** [comment]
Thanks Annie, I do think this is what this means, especially if the location is the only thing you changed. For NAIP data I think it’s possible they only cover the continental US. You could consider looking at Landsat or Sentinel 2 data instead for your portfolio. I will add some code to address this situation in the future.

**Reply by annieM459** [reply]
Great info! Thanks Elsa!

**Comment by peterkobylarz** [comment]
@annieM459 I ran into the same issue using a different shapefile as my input for my portfolio project. My shapefile was using a state plane coordinate system, so the bounding box being sent to the downloader was in feet rather than latitude/longitude and no NAIP datasets were being found. When I converted the CRS to WGS 1984 (ESPG 4326), it fixed the error. This might be worth trying for you.

```
oak_gdf = gpd.read_file(oak_path).set_index('NAME').to_crs(4326)
```

**Reply by annettebalmes** [reply]
This is what worked in my case!
